topics/lmms/rqtl2/gn-rqtl-design-implementation.gmi

```diff
@@ -0,0 +1,118 @@
+# RQTL Implementation for GeneNetwork Design Proposal
+
+## Tags
+
+* Assigned: alexm,
+* Keywords: RQTL, GeneNetwork2, Design
+* Type: Enhancements,
+* Status: In Progress
+
+
+
+## Description
+
+This document outlines the design proposal for the re-implementation of the RQTL feature in GeneNetwork providing also  a console view to track the external process.
+
+### Problem Definition
+The current RQTL implementation requires enhancements. The core functionality remains the same: making API calls from GeneNetwork 3 (GN3). The system needs a cleaner architecture for future use, emphasizing improved error handling and a clear separation of concerns between GeneNetwork 2 (GN2) and GN3. This will involve eliminating file transfers between GN2 and GN3. Additionally, a console should be provided to users for tracking the progress of ongoing tasks.
```

Comment:
Can we split up the challenges and features we want to introduce? Something like:
```
The current RQTL implementation has the challenges:

- challenge1
- challenge2

We'll fix the challenges and add enhancements by:

- adding a console ...
- eliminating file transfers ...
```

Reply:
this commit addresses this issue: 
https://github.com/genenetwork/gn-gemtext-threads/pull/46/commits/6f17825e7783ffd3b2d67306b6885d9e061fd313